src/panels/config/script/ha-script-editor.ts

```diff
@@ -830,7 +719,15 @@ export class HaScriptEditor extends KeyboardShortcutMixin(LitElement) {
       return;
     }
 
+    if (!this.scriptId) {
+      const saved = await this._promptAutomationAlias();
+      if (!saved) {
+        return;
+      }
+      this.updateEntityId(undefined, this._config!.alias);
```

Comment:
We can remove the `updateEntityId` method as it will always be generated using the `_computeEntityIdFromAlias` method.